lisa/tools/mde.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+import sys
+import json
+import requests
+from pathlib import Path, PurePath
+from typing import Any
+
+from lisa.executable import Tool
+from lisa.executable import CustomScriptBuilder, CustomScript
+#from . import RemoteCopy, Whoami, Curl
+from .remote_copy import RemoteCopy
+from .whoami import Whoami
+
+class MDE(Tool):
+    @property
+    def command(self) -> str:
+        return "mdatp"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def get_mde_installer(self) -> bool:
+        if not hasattr(self, '_mde_installer'):
+            response = requests.get("https://raw.githubusercontent.com/microsoft/mdatp-xplat/master/linux/installation/mde_installer.sh")
+            if response.ok:
+                script = response.text
+                import tempfile
+                _, self.mde_installer_script = tempfile.mkstemp(prefix='mde_installer', suffix='.sh')
+                with open(self.mde_installer_script, 'w') as writer:
+                    writer.write(script)
+                self._mde_installer = CustomScriptBuilder(Path(os.path.dirname(self.mde_installer_script)),
+                                                [os.path.basename(self.mde_installer_script)])
+                return True
+            return False
+        return True
+
+    def _install(self) -> bool:
+        if not self.get_mde_installer():
+           self._log.error("Unable to download mde_installer.sh script. MDE can't be installed")
+
+        mde_installer: CustomScript = self.node.tools[self._mde_installer]
+        self._log.info('Installing MDE')
+        result1 = mde_installer.run(parameters="--install", sudo=True)
+        self._log.info(result1)
+
+        return self._check_exists()
+
+
+    def onboard(self, onboarding_script: PurePath) -> bool:
+        if not self._check_exists():
+            self._log.error("MDE is not installed, onboarding not possible")
+            return False
+
+        username = self.node.tools[Whoami].get_username()
+
+        remote_copy = self.node.tools[RemoteCopy]
+        remote_copy.copy_to_remote(onboarding_script, PurePath(f"/home/{username}/MicrosoftDefenderATPOnboardingLinuxServer.py"))
```

Comment:
we can download the file on the vm directly, no need remote copy from the running machine?